pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -893,9 +886,17 @@ func (c *Reconciler) createRunObjects(ctx context.Context, rpt *resources.Resolv
 	var runObjects []v1beta1.RunObject
 	ctx, span := c.tracerProvider.Tracer(TracerName).Start(ctx, "createRunObjects")
 	defer span.End()
-	matrixCombinations := rpt.PipelineTask.Matrix.FanOut()
+	var matrixCombinations []v1beta1.Params
+
+	if rpt.PipelineTask.IsMatrixed() {
+		matrixCombinations = rpt.PipelineTask.Matrix.FanOut()
+	}
+
 	for i, runObjectName := range rpt.RunObjectNames {
-		params := matrixCombinations[i]
+		var params v1beta1.Params
+		if rpt.PipelineTask.IsMatrixed() {
+			params = matrixCombinations[i]
+		}
```

Comment:
I'd expect this to be the same as L815 - L816

```suggestion
		var params v1beta1.Params
		if len(matrixCombinations) > i {
			params = matrixCombinations[i]
		}
```